--- at_response.c.orig	2026-02-16 07:20:53.000000000 +0000
+++ at_response.c	2026-02-16 07:24:51.995745423 +0000
@@ -1255,6 +1255,7 @@
 	char scts[64], dt[64];
 	int mr, st;
 	char		msg[4096];
+	char cmgr_raw[8192];
 	int		res;
 	char		text_base64[40800];
 	size_t msg_len = sizeof(msg);
@@ -1275,6 +1276,7 @@
 	if (ecmd) {
 		if (ecmd->res == RES_CMGR || ecmd->cmd == CMD_USER) {
 			at_queue_handle_result (pvt, RES_CMGR);
+		ast_copy_string(cmgr_raw, str, sizeof(cmgr_raw));
 
 			res = at_parse_cmgr(str, len, &tpdu_type, sca, sizeof(sca), oa, sizeof(oa), scts, &mr, &st, dt, msg, &msg_len, &udh);
 			if (res < 0) {
@@ -1343,6 +1345,7 @@
 					channel_var_t vars[] =
 					{
 						{ "SMS", fullmsg } ,
+			{ "CMGR", cmgr_raw },
 						{ "SMS_BASE64", text_base64 },
 						{ "SMS_TS", scts },
 						{ NULL, NULL },
